Python/Product/PythonTools/PythonTools/Repl/PythonDebugReplEvaluator.cs

```diff
@@ -421,15 +410,38 @@ internal class PythonDebugReplEvaluator :
                 return;
             }
 
+            // Multiple callers can execute in this method simultaneously (due to async/await)
+            // so we need to protect against 2 callers trying to create evaulators for the same process.
+
+            // Keep track of evaluators that are in progress of attaching, and if
+            // we are getting called to attach for one that is already in progress,
+            // just wait for it to finish before returning.
+            Task attachingTask;
+            TaskCompletionSource<object> attachingTcs = null;
+            if (_attachingTasks.TryGetValue(process.Id, out attachingTask)) {
```

Comment:
Need a lock here to avoid races (two threads check, then both add the task).

Concurrent dict doesn't solve this - it doesn't have a function for "get it if it's there, create it if not, **and tell me whether you just created it**".